build-info-extractor/src/main/java/org/jfrog/build/extractor/issuesCollection/IssuesCollector.java

```diff
@@ -141,6 +141,11 @@ private String getGitLog(File execDir, Log logger, String previousVcsRevision) t
         CommandExecutor commandExecutor = new CommandExecutor("git", null);
         CommandResults res = commandExecutor.exeCommand(execDir, args, null, logger);
         if (!res.isOk()) {
+            Pattern pattern = Pattern.compile("fatal: Invalid revision range [a-fA-F0-9]+\\.\\.");
```

Comment:
Since the ```Pattern.compile``` action is expensive, we'd like to avoid executing it everytime. Let's compile the pattern once, the first time it is needed and save it. When implementing this change, let's also avoid compiling the pattern, if it is not used.